Design updates for #2015 numeric type literal syntax (#2410)
carbon-language/carbon-lang: docs/design/expressions/README.md

`if` expressions are analogous to `?:` ternary expressions in C and C++.

> ## Numeric type literals expressions

Reviewer: ```suggestion
## Numeric type literal expressions
```

Carbon's syntax provides a simple way to represent different types of integers
and floating-point numbers. Each type is identified with a keyword-like syntax,
prefixed with either `i`, `u`, or `f` followed by a multiple of 8, representing

> the size of the data type.

Reviewer: ```suggestion
the size in bits of the data type.
```

> The types referred to as [Numeric type literals expression](literals.md).

Reviewer 2: ```suggestion
The types are referred to as [numeric type literals expressions](literals.md).
```

> The types referred to as [Numeric type literals expression](literals.md).

Reviewer: ```suggestion
These are referred to as
[numeric type literals](literals.md#numeric-type-literals).
```
